### ISSUE TITLE
Articles widget

### ISSUE THREAD

**Issue by jocmp** [jocmp/capyreader#214]
As a user, I can see a list of the latest articles.

### Reference

- https://developer.android.com/develop/ui/compose/glance/create-app-widget

**Comment by theolm** [comment]
Hello there. Do you accept contributions for the project?
If so, any guidelines that I should follow or just the regular fork -> branch -> PR?

Btw, I'm asking here because I'm interested in working on the widget.

**Comment by jocmp** [comment]
@theolm, yep! If you have something in mind I'll take contributions for the widget. And you're right about branching. Fork this repo, and open a PR. Let me know if I can answer any questions about the codebase along the way.

**Comment by theolm** [comment]
Nice. I'll work on the widget over the weekend.
I'll not use Glance though, the API is very limited, and basically impossible to build anything decent.
You can't have a scroll for example (which I believe is important for this widget).

**Comment by jocmp** [comment]
Thanks! And good call about not using Glance. I tried Glance early on when I tried to add this feature and it didn't work as expected.

**Comment by dubsflyt** [comment]
Would love a widget design similar to palabre here where it shows the latest article (or a few and occasionally rotates between them)

![Image](https://github.com/user-attachments/assets/6262d29f-2c06-40a6-a460-04d209e3fcfd)

**Comment by jocmp** [comment]
@dubsflyt thanks for the example! My first goal is a list design, but I'm also a fan of this carousel design too. I'll look into it as well.

**Comment by theolm** [comment]
Sorry for the delay on this one. I'm very busy at work and I don't have much time to work on side projects.
I already started the implementation. I have the widget/ui ready, I just need to hook up the service that fetch the articles.

@jocmp, what would be the best way to fetch the articles? Ideally, I would call some service to update and then read from local-storage (shared pref maybe). Could you give me some directions?

**Comment by jocmp** [comment]
Hey @theolm - no problem, I appreciate the willingness to help.

I started a Jetpack Glance implementation of this that I'm close to wrapping up myself. It uses the `account.latestArticles` here

- https://github.com/jocmp/capyreader/pull/1018

Definitely feel free to open a pull request with anything you have done up to this point even if it's still work in progress. I might be able to hook up the networking code myself or combine these two branches.

**Comment by dubsflyt** [comment]
> [@dubsflyt](https://github.com/dubsflyt) thanks for the example! My first goal is a list design, but I'm also a fan of this carousel design too. I'll look into it as well.

The new widget in the new version is great - is it possible to do a version of the widget that just shows the top 2 articles as they show now with no scrolling?
Sort of a mid way point between my suggestion and what is there now!

**Comment by jocmp** [comment]
That might work - is there a benefit to having a static list of the latest two headlines instead of a scrolling list? I'm not a heavy widget user so I'm curious about the use cases.

**Comment by jocmp** [comment]
Perfect, I'll start start tracking the carousel widget with a new issue, https://github.com/jocmp/capyreader/issues/1065. It will be nice to add a different widget with images.

I'll keep the door open for a non-scrolling widget though since I have seen this pattern on other widgets. Like you said, it's definitely more glanceable to have a static list.
